### PR TITLE
[workload] Finish with failure on admission check rejection.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Finish the workload with failure if any of its admission checks is marked as Rejected.

#### Which issue(s) this PR fixes:

Relates to  #1136 

#### Does this PR introduce a user-facing change?

```release-note
Finish the workload with failure if any of its admission checks is marked as Rejected.
```

### CODE REVIEW

File: pkg/workload/workload.go [hunk 1]
```diff
@@ -454,6 +454,18 @@ func HasRetryOrRejectedChecks(wl *kueue.Workload) bool {
 	return false
 }
 
+// Returns the list of Rejected admission checks
+func GetRejectedChecks(wl *kueue.Workload) []string {
```
Comment (mimowo) [inline pkg/workload/workload.go:458]: I would suggest placing this function is some utility package for admission checks.

Comment (trasc) [reply pkg/workload/workload.go:458]: Let's create the utility package with, or after #1174

File: pkg/controller/core/workload_controller.go [hunk 2]
```diff
@@ -134,6 +135,17 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 {
```
Comment (alculquicondor) [inline pkg/controller/core/workload_controller.go:138]: This is good by itself, but we should also make sure that a job cannot be unsuspended when the workload is finished.
If it's not already the case, you can do it in a follow up.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
NONE
```
Since this is a follow up to #1045.

**Pull request merged**